scripts/python/verify_agent_os.py

```diff
@@ -0,0 +1,14 @@
+# Script verifies the operating system for the platform on which it is being run
+# Used in build pipelines to verify the build agent os
+# Variable: The name of the os to verfy against
+
+import sys
+import platform
+osparameter = sys.argv[1]
+print ("Parameter passed by matrix -", osparameter)
+osparameter = 'Darwin' if(osparameter.lower() == 'MacOS'.lower()) else osparameter
+agentos = platform.system()
+if (agentos.lower().startswith(osparameter.lower()[:3])):
```

Comment:
Is there a reason you did a startswith as opposed to exact equality?

Reply:
Also why are you comparing on the first 3 characters?

Reply:
Oh, I must have carried that over from when I was using `sys.platfrom` which gives more specific values like `win32`, `win64`, or `linux2`. Also I was thinking that there could be cases where a more specific value is passed to the script as osparameter.